src/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -411,5 +411,53 @@ private static unsafe int WriteFileNative(IntPtr hFile, byte[] bytes, int offset
                 return errorCode;
             }
         }
+
+        internal sealed class ControlCHandlerRegistrar
+        {
+            private bool _handlerRegistered;
+            private Interop.mincore.ConsoleCtrlHandlerRoutine _handler;
+
+            internal ControlCHandlerRegistrar()
+            {
+                _handler = new Interop.mincore.ConsoleCtrlHandlerRoutine(BreakEvent);
+            }
+
+            internal void Register()
+            {
+                if (!_handlerRegistered)
```

Comment:
In what situation would this be called when _handlerRegistered is true?  I may have just misunderstood how this is used, but it seemed like this if could be removed and replaced with an assert.